frontend/src/app/modules/lessons/lesson/lesson.component.ts

```diff
@@ -65,39 +69,83 @@ export class LessonComponent extends BaseComponent implements OnInit {
-            this.questions = questions as Question[];
+            this.questions = questions;
             this.isLoading = false;
             this.spinner.hide();
         });
     }
 
-    enrollLesson() {
-        this.userService.enrollUserToLesson(this.lesson.id)
+    joinLesson() {
+        this.userService
+            .enrollUserToLesson(this.lesson.id)
             .pipe(this.untilThis)
-            .subscribe({ next: (lesson) => {
-                this.lesson.isSubscribed = true;
-                this.lesson.subscribersCount = lesson.subscribersCount;
-                this.notificationService.showSuccess(`You successfully registered for lesson ${this.lesson.title}`, 'Success!');
-            },
-            error: () => this.notificationService.showError(`Failed to register for lesson ${this.lesson.title}`, 'Failed!') });
+            .subscribe({
+                next: (lesson) => {
+                    this.lesson.isSubscribed = true;
+                    this.lesson.subscribersCount = lesson.subscribersCount;
+                    this.notificationService.showSuccess(
+                        `You successfully registered for lesson ${this.lesson.name}`,
+                        'Success!',
+                    );
+                },
+                error: () =>
+                    this.notificationService.showError(`Failed to register for lesson ${this.lesson.name}`, 'Failed!'),
+            });
+    }
+
+    cancelLesson() {
+        this.lessonsService.cancelLesson(this.lesson.id, this.lesson).subscribe((data) => {
+            this.lesson.isCanceled = data.isCanceled;
+        });
     }
 
     ngOnInit(): void {
-        this.previewImage = this.lesson.videoId ? `//img.youtube.com/vi/${this.lesson.videoId}/maxresdefault.jpg` : '';
+        this.previewImage = this.lesson.youtubeVideoId
+            ? `//img.youtube.com/vi/${this.lesson.youtubeVideoId}/maxresdefault.jpg`
+            : '';
     }
 
     isDisabled() {
-        return this.lesson.isSubscribed || new Date() > this.lesson.startAt;
+        if (!this.isTeachersPage) {
+            return this.lesson.isSubscribed || new Date() > new Date(this.lesson.startAt);
+        }
+        if (this.isTeachersPage) {
+            return this.lesson.isCanceled || new Date() > new Date(this.lesson.startAt);
+        }
+
+        return false;
     }
 
     getButtonContent() {
-        if (new Date() > this.lesson.startAt) {
-            return 'Expired';
+        if (!this.isTeachersPage) {
+            if (new Date() > new Date(this.lesson.startAt)) {
+                return 'Expired';
+            }
+            if (this.lesson.isSubscribed) {
+                return 'Subscribed';
+            }
+
+            return 'Join';
         }
-        if (this.lesson.isSubscribed) {
-            return 'Subscribed';
+
+        if (this.isTeachersPage) {
+            if (!this.lesson.isCanceled) {
+                return 'Cancel';
+            }
+
+            return 'Canceled';
```

Comment:
What about switch/case? Like switch(true) { case this.isTeachersPage: return 'Canceled' } and so on